### PR TITLE
Apply custom message in login dialog on direct access

### PR DESCRIPTION
**Pull request by knolleary**
Fixes https://github.com/FlowFuse/node-red/issues/25

This PR removes the username/password fields from the login dialog when the editor is accessed directly. We get support requests asking what credentials to use - however we only support accessing the editor via the device agent proxy.

With NR 4.0.6 (to be released imminently), I've added the ability to insert a custom message/button on the login dialog when in this state. Older versions of NR will show a blank dialog.

image

Note: spotted the `dashbaord` typo when I took this screenshot for the PR... it will be fixed momentarily.

### CODE REVIEW

File: lib/template/template-settings.js [hunk 1]
```diff
@@ -169,6 +162,15 @@ const runtimeSettings = {
     [themeName]: { ...themeSettings },
     editorTheme: { ...editorTheme }
 }
+runtimeSettings.editorTheme.login = {
+    message: 'Access to the editor is managed through the FlowFuse platform'
+}
+if (themeSettings.projectURL) {
+    runtimeSettings.editorTheme.login.button = {
+        url: themeSettings.projectURL,
+        label: 'Open FlowFuse Dashbaord'
```
Comment (knolleary) [inline lib/template/template-settings.js:171]:
```suggestion
        label: 'Open FlowFuse Dashboard'
```

**Comment by Steve-Mcl** [comment]
@knolleary not sure if you noticed test fail following removal of `type`?

**Comment by knolleary** [comment]
@Steve-Mcl test fix pushed

**Review by Steve-Mcl — APPROVED** [review]
Tested/working/approved

This is how it looks on Win + Chrome

image

Aesthetically, I do wish the string didn't wrap on the last word (but at the same time I cant think of a better way to phrase it)

**Comment by Steve-Mcl** [comment]
> Aesthetically, I do wish the string didn't wrap on the last word (but at the same time I cant think of a better way to phrase it)

Perhaps `'Access to the editor is managed via FlowFuse'` would be a suitable (shorter, non wrapping) alternative?

File: lib/template/template-settings.js [hunk 2]
```diff
@@ -169,6 +162,15 @@ const runtimeSettings = {
     [themeName]: { ...themeSettings },
     editorTheme: { ...editorTheme }
 }
+runtimeSettings.editorTheme.login = {
+    message: 'Access to the editor is managed through the FlowFuse platform'
```
Comment (knolleary) [inline lib/template/template-settings.js:166]:
```suggestion
    message: 'Access the editor through the FlowFuse platform'
```

**Comment by knolleary** [comment]
Have updated the message to `Access the editor through the FlowFuse platform`

**Pull request merged**